warehouse/warehouse.go

```diff
@@ -76,10 +76,11 @@ var (
 
 // warehouses worker modes
 const (
-	MasterMode      = "master"
-	SlaveMode       = "slave"
-	MasterSlaveMode = "master_and_slave"
-	EmbeddedMode    = "embedded"
+	MasterMode        = "master"
+	SlaveMode         = "slave"
+	MasterSlaveMode   = "master_and_slave"
+	EmbeddedMode      = "embedded"
+	PooledWHSlaveMode = "pooled_wh_slave"
```

Comment:
@AkhileshK8y Are we always going to run wh master on core server when using pooled slaves (core+pooled_slaves). Shouldn't we always support running it as core+master+pooled_slaves

Reply:
core+master+pooled_slaves <- this is embedded mode only right? 

In multi-tenant version, we will be running slaves as separate service and warehouse master as part of backend. 

@AkhileshK8y Let's call this new mode `embedded_master` ?

Reply:
@psrikanth88 I assume what you mean is why not separate master when we are having separate pool... Yes for now we will always run wh master inside core server.. which @snarkychef  is calling embedded_master mode..

Reply:
@snarkychef @AkhileshK8y  yeah, I meant separate master with slaves pooled. Cool, if we are not going to use it like that its fine.